pkg/datalayer/datalayer.go

```diff
@@ -21,10 +22,50 @@ func VolumeExists(basePath string, volumeName string) bool {
 	return err == nil
 }
 
+func ActiveVolume(basePath string) (string, error) {
+	currentVolumeJsonPath := filepath.FromSlash(basePath + "/" + "current_volume.json")
+	file, err := os.Open(currentVolumeJsonPath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
```

Comment:
I *think* you're meant to put `defer` statements before `return` statements. IOW, as soon as you open the file, defer closing it to whenever the function stops executing.